firewallrule.go

```diff
@@ -64,11 +78,14 @@ func (c *Client) CreateFirewallRule(dcID string, serverID string, nicID string,
 	return ret, err
 }
 
-//UpdateFirewallRule updates a firewall rule
+// UpdateFirewallRule updates a firewall rule.
+// You need to pass all wanted properties, not just those you want to change.
 func (c *Client) UpdateFirewallRule(dcID string, serverID string, nicID string, fwID string, obj FirewallruleProperties) (*FirewallRule, error) {
+	props := obj.DeepCopy()
+	props.Protocol = "" // the field is immutable and must not be sent in an update request
```

Comment:
I don't think that this should be handled here, as users might wonder why they cannot change the protocol. Just escalate the API error to the user. Eventually this also makes the FirewallRules.DeepCopy method nice but unnecessary. As this would be the only struct with this option I'd say drop it too.